ci/check-if-docs-pr.sh

```diff
@@ -24,14 +24,31 @@ fi
 
 # List the files that are touched in the PR, filter for docs changes and remove
 # whitespace. If output is empty then only docs are changed.
-# Note: known limitation, see
-# opstrace-prelaunch/issues/1708#issuecomment-717404504
-
-# For debugging, how what this command really returns
-# git --no-pager diff HEAD main
-git --no-pager diff --name-only HEAD main
-
-DOCS_ONLY_CHANGES=$(git --no-pager diff --name-only HEAD main | egrep -v "^docs/|.gitattributes|^.github|^README.md|^.markdownlint.json" | tr -d '[:space:]')
+#
+# https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#list-pull-requests-files
+FILES_EDITED_IN_PR=$(
+    curl \
+    --silent \
+    -H "Accept: application/vnd.github.v3+json" \
+    https://api.github.com/repos/opstrace/opstrace/pulls/${BUILDKITE_PULL_REQUEST}/files \
```

Comment:
How much do we know about the consistency of the HTTP API with the actual code? Hm. :) We'll find out, I guess.

Reply:
> How much do we know about the consistency of the HTTP API with the actual code? 

Mind clarifying your question please?

BTW, don't know if you noticed but I left a link to the [github API documentation](https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#list-pull-requests-files) in the code comments.

Reply:
> Mind clarifying your question please?

Yes! :-)

When we send the GET to `https://api.github.com/repos/opstrace/opstrace/pulls/${BUILDKITE_PULL_REQUEST}/files` then I suppose the HTTP response is mean to reflect the _current_ state of the pull request. That is, the response is sensitive to _timing_. The URL, so to say, is a moving target. The request does not contain commit information.

A challenging scenario might be two git pushes happening "at the same time" to the same branch, triggering a build quasi-simultaneously.

While that might be an extreme scenario.. what if the two pushes are 5 seconds apart, 10 seconds, 1 minute?

I suppose we _hope_ that in practice there are no mean race conditions here where `https://api.github.com/repos/opstrace/opstrace/pulls/${BUILDKITE_PULL_REQUEST}/files` maybe sometimes reflects the state of the pull request from 30 seconds ago, instead of having already picked up the last push from 1 second ago.

Reply:
If this ever becomes a problem: the JSON response seems to contain a `sha` property which we can compare to the current checkout maybe!